### ISSUE TITLE
Add new custom parameters to script commands - for example "EJS_Shader = 4xScaleHQ"

### ISSUE THREAD

**Issue by shaylevary** [emulatorjs/emulatorjs#290]
Hi,

It will be nice if you can you add some more custom parameters to script - for example:
EJS_Shader = 4xScaleHQ ;
EJS_Cheat = true;
EJS_Frameskip = 4;
EJS_EnhancedResLow = 0;
EJS_EnhancedResHigh = 1;

ect...

Thank you,
Shay.

**Comment by ethanaobrien** [comment]
I can’t do this. It wouldn’t even be possible to set the cheats to true, as not all emulators support it, and showing the button won’t do anything. The only one in your example that makes sense is the shaders, you can do this by setting the localstorage variable with that set as. (Start a game, turn on that shader, view the localstorage objects, etc…). 
Frameskip also makes sense and can be done the same way, but is only supported on the ds core.
You can’t change the resolution. It’s simply not possible, sorry.

If you have any questions about using localstorage to archive the pre-set settings, please let me know

**Comment by shaylevary** [comment]
About the cheat - ok.
About the shaders and the frameskip - i will appreciate if you can please help me on how to use the localstorage in order to update this settings.

btw: PSX games support frameskip and it sometimes helps with the lags.

PS - I've opened another ticket on the emuserver project, i couldn't compile the project - npm is missing some packages, i will really appreciate if you can please help with this issues.

**Comment by shaylevary** [comment]
Hi,

About the cheat - ok, I understand.


About the shaders and the frameskip - i will appreciate if you can please
help me on how to use the localstorage in order to update this settings.

btw: PSX games support frameskip and it sometimes helps with the lags.

PS - I've opened another ticket on the emuserver project, i couldn't
compile the project - npm is missing some packages, i will really
appreciate if you can please help with this issues.


Thank you.


בתאריך יום ג׳, 12 ביולי 2022, 16:40, מאת ethanaobrien ‏<
***@***.***>:

**Comment by ethanaobrien** [comment]
Using this option is going to be much better to force default settings, as the users preferences will save and take priority. If you used local storage, all user settings will be re-written. Let me know if you still need these instructions.

All core options are displayed in the settings menu. They are taken straight from the core to make it easier if they add/remove anything.

The emuserver project has been private for months... I'm in the process of completely re-writing netplay using Microsoft dot net C#. The eta on this is probably 2-4 months.

Sorry I haven't been as active. I have a real job now and that takes up a lot of time

**Comment by shaylevary** [comment]
Yes, I will appreciate if you can please help when you'll have time with instructions of how to use the local storage in order to overwrite user settings.

Regarding the project, as I have already mentioned to you before, I really appreciate the amazing work you are doing here.
Looking forward for the new dot-net netplay support, I am currently using the latest version that support netplay, it works great for me, the only drawback is that i'm looking for a way to enter a pause command in the script code in order to pause the game when the user presses the netplay button.
Only after player 2 joins the game - the script should send a PLAY command for both players (in this way the frame-sync is good for both players and the same content is shown for both of them simultaneously on the screen).

I wish you good luck with the new job, take all the time you need to develop this project on your free time.
